### PR TITLE
wayland: Fix toggling fullscreen with fixed-size windows

### LINKED ISSUE #4281: [Wayland] Fullscreen toggles get canceled out by a bogus resize
Pretty easy test: Open a 1280x720 window, then call SDL_SetWindowFullscreen with SDL_WINDOW_FULLSCREEN_DESKTOP. At first it does the usual resize event to your desktop resolution, but then it comes right back around and gets resized to the old windowed mode of 1280x720. This is presumably because of the `pending` value causing an outdated resize to get sent after the fullscreen set has been done. The result is bogus window information, which means your eventual glBlitFramebuffer will eventually be wrong! OpenGL and Vulkan behave differently because of HandlePendingResize timing differences, but they're both wrong in their own strange way.

Interestingly the current libdecor path in #4068 is unaffected by this, but it may also explain why the window size is completely wrong when going from windowed -> fullscreen -> windowed, so I'll CC @christianrauch here just in case.

Capsized is my test case for this since it has hardcoded screen size values (sorry), it is on the very latest FNA as of writing.

### PR DESCRIPTION
**Pull request by flibitijibibo**
Fixes #4281.

Continuing off my work from #4065, it looks like Wayland took even _more_ notes from X and made the min/max an issue when calling the set_fullscreen functions, causing the display server to clamp the window and resize it to something really weird and wrong. The solution was to fold _all_ of the min/max work into a new function and do a full commit every single time we change the dimensions in some way. It turns out it's cleaner like this anyway!

### CODE REVIEW

File: src/video/wayland/SDL_waylandwindow.c [hunk 1]
```diff
@@ -43,12 +43,63 @@ static float get_window_scale_factor(SDL_Window *window) {
...
 
+static void
+CommitMinMaxDimensions(SDL_Window *window)
+{
+    SDL_WindowData *wind = window->driverdata;
+    SDL_VideoData *data = wind->waylandData;
+    int min_width, min_height, max_width, max_height;
+
+    /* FIXME: Is there a better way to get max window size from Wayland? -flibit */
+    const int maxsize = 0x7FFFFFFF;
```
Comment (christianrauch) [inline src/video/wayland/SDL_waylandwindow.c:54]: If you want to tell the Wayland compositor/server not to constrain the maximum size, just send 0: `xdg_toplevel_set_max_size(..., 0, 0)`. There is no need to figure out the maximum size.

File: src/video/wayland/SDL_waylandwindow.c [hunk 2]
```diff
@@ -43,12 +43,63 @@ static float get_window_scale_factor(SDL_Window *window) {
...
+        zxdg_toplevel_v6_set_min_size(wind->shell_surface.zxdg.roleobj.toplevel,
+                                      min_width,
+                                      min_height);
+        zxdg_toplevel_v6_set_max_size(wind->shell_surface.zxdg.roleobj.toplevel,
+                                      max_width,
+                                      max_height);
+    }
+
+    wl_surface_commit(wind->surface);
+    WAYLAND_wl_display_flush(data->display);
```
Comment (christianrauch) [inline src/video/wayland/SDL_waylandwindow.c:89]: The "flush" is not required. Only the "commit" is required to tell the compositor that you are done with setting a new state.

**Review by christianrauch — CHANGES_REQUESTED** [review]

**Comment by flibitijibibo** [comment]
Pushed an update per review comments from @christianrauch, thanks for the tips!

**Comment by flibitijibibo** [comment]
Pushed an additional commit that fixes this for Plasma as well. The behavior was radically different because of an unassigned wl_output variable, which _should_ have affected every Wayland environment but mysteriously only affected Plasma.

**Pull request merged**